### ISSUE TITLE
Urls with hashes are not correctly processed in js

### ISSUE THREAD

**Issue by 40rn05lyv** [aramrami/OWASP-CSRFGuard#40]
`injectTokenAttribute` in js just concats existing several strings and doesn't process cases, when url contains hash part (http://localhost/context/asdf?query#hash)

**Solution:**

I added following code (see http://stackoverflow.com/questions/486896/adding-a-parameter-to-the-url-with-javascript):

```
    function setParameter(url, key, value) {
        key = encodeURIComponent(key);
        value = encodeURIComponent(value);
        var parser = document.createElement('a');
        parser.href = url;
        var kvp = parser.search.substr(1).split('&');
        var newSearch = '';
        if (kvp == '') {
            newSearch = '?' + key + '=' + value;
        } else {
            var i = kvp.length;
            var x; 
            while (i--) {
                x = kvp[i].split('=');
                if (x[0] == key) {
                    x[1] = value;
                    kvp[i] = x.join('=');
                    break;
                }
            }
            if (i < 0) { 
                kvp[kvp.length] = [key, value].join('=');
            }
            newSearch = kvp.join('&');
        }
        parser.search = newSearch;
        return parser.href;
    }

```

and changed from

```
			if(location.indexOf('?') != -1) {
				location = location + '&' + tokenName + '=' + value;
			} else {
				location = location + '?' + tokenName + '=' + value;
			}
```

to

```
location = setParameter(location, tokenName, tokenValue)
```

**Comment by 40rn05lyv** [comment]
This code will also fix #31

**Comment by aramrami** [comment]
Please send a Pull Request.
